app/models/commercial.rb

```diff
@@ -1,10 +1,43 @@
 class Commercial < ActiveRecord::Base
+  require 'oembed'
+
   belongs_to :commercialable, polymorphic: true
 
-  attr_accessible :commercial_id, :commercial_type
+  attr_accessible :url
+
+  validates :url, presence: true
+  validates :url, format: URI::regexp(%w(http https))
+
+  def self.provider
```

Comment:
This is only for backward compatibility because I don't drop the old fields in the DB! If there is a way to see the registered providers, I don't think that they correspond with the value we stored in the DB! Anyway, this is only for backward compatibility, so I think dropping this method and hardcode in the view should be OK. We needed the function for the drop down menu of the old functionality...